presto-main/src/main/java/io/prestosql/operator/scalar/TryCastFunction.java

```diff
@@ -76,20 +75,20 @@ public ScalarFunctionImplementation specialize(FunctionBinding functionBinding,
         Type fromType = functionBinding.getTypeVariable("F");
         Type toType = functionBinding.getTypeVariable("T");
 
-        Class<?> returnType = Primitives.wrap(toType.getJavaType());
+        Class<?> returnType = wrap(toType.getJavaType());
 
         // the resulting method needs to return a boxed type
-        MethodHandle coercion = functionDependencies.getCastInvoker(fromType, toType, Optional.empty()).getMethodHandle();
+        InvocationConvention invocationConvention = new InvocationConvention(ImmutableList.of(NEVER_NULL), NULLABLE_RETURN, true, false);
+        MethodHandle coercion = functionDependencies.getCastInvoker(fromType, toType, invocationConvention).getMethodHandle();
         coercion = coercion.asType(methodType(returnType, coercion.type()));
 
         MethodHandle exceptionHandler = dropArguments(constant(returnType, null), 0, RuntimeException.class);
         MethodHandle tryCastHandle = catchException(coercion, RuntimeException.class, exceptionHandler);
 
-        boolean nullableArgument = functionDependencies.getCastMetadata(fromType, toType).getArgumentDefinitions().get(0).isNullable();
         return new ChoicesScalarFunctionImplementation(
                 functionBinding,
                 NULLABLE_RETURN,
-                ImmutableList.of(nullableArgument ? BOXED_NULLABLE : NEVER_NULL),
+                ImmutableList.of(NEVER_NULL),
```

Comment:
why no longer nullable?

Reply:
It doesn't need to be nullable.  If the argument is null, the engine will automatically return null for you.